CONTRIBUTING.md

```diff
@@ -0,0 +1,67 @@
+# Contributing
+
+Thanks a lot for your interest in contributing to **editorconfig-checker**!
+
+## Types of contributions
+
+Contributions to **editorconfig-checker** _include_, but is _not limited to_:
+
+- Report a bug.
+- Suggest a new feature idea.
+- Correct spelling errors, improvements, or additions to documentation files (README, CONTRIBUTING...).
+- Improve structure/format/performance/refactor/tests of the code.
+
+All contributions are welcome from anyone willing to work in good faith with other contributors and the community. No contribution is too small and all contributions are valued.
+
+## Open Development
+
+All work on **editorconfig-checker** happens directly on [GitHub](https://github.com/editorconfig-checker/editorconfig-checker). Both core team members and external contributors send pull requests, which go through the same review process.
+
+## Pull Requests
+
+Pull Requests are the way concrete changes are made to the code and the documentation.
+
+**Please first discuss** the change you wish to make via [issue](https://github.com/editorconfig-checker/editorconfig-checker/issues) before making a change. It might save you time.
```

Comment:
I'm not a fan of requiring an issue before a PR. It greatly increases the effort required for a contributor to make a simple contribution.

Reply:
I agree. If there is a need for an issue, we can ask the contributor to open one.

I often make PRs for fixing typos, requiring to open a PR for such PR would be counterproductive

Reply:
Yes, not every PR needs to have first an issue open, I agree. I think the wording is misleading here, it was more a recommendation, that opening an issue first allows discussions and check if your goals match the goal of everyone else, before implementing something not wanted and "wasting time".

Reply:
Maybe a wording like "please discuss bigger changes first" could make it clearer, that we encourage quick and simple PRs.

Reply:
Resolved by [`923493b` (#378)](https://github.com/editorconfig-checker/editorconfig-checker/pull/378/commits/923493be7d65693b0d2cb71122ef95957da53589)